walk/git.go

```diff
@@ -3,16 +3,71 @@ package walk
 import (
 	"context"
 	"fmt"
+	"github.com/charmbracelet/log"
+	"github.com/go-git/go-git/v5/plumbing/filemode"
+	"github.com/go-git/go-git/v5/plumbing/format/index"
 	"io/fs"
 	"os"
 	"path/filepath"
-
-	"github.com/charmbracelet/log"
+	"strings"
 
 	"github.com/go-git/go-git/v5"
-	"github.com/go-git/go-git/v5/plumbing/filemode"
 )
 
+// fileTree represents a hierarchical file structure with directories and files.
+type fileTree struct {
```

Comment:
Simple tree structure for checking paths exist, be they real files or directories, in the git index.